docs/tutorial.rst

```diff
@@ -0,0 +1,144 @@
+Tutorial
+========
+To get started, let's first install the TensorNetwork library.
+
+.. code-block:: bash
+
+  pip3 install tensornetwork
+
+
+Nodes
+-----
+Nodes are one of the basic building blocks of a tensor network. They represent a tensor in the computation. Each axis will have a corresponding edge that can possibly connect different nodes (or even the same node) together. The number of edges represents the order of the underlying tensor. For example, a node without any edges is a scalar, a node with one edge is a vector, etc.
+
+.. code-block:: python
+
+  import tensornetwork
+  import tensorflow as tf
+
+  net = tensornetwork.TensorNetwork()
+  node = net.add_node(tf.eye(2)) # NumPy arrays can also be used.
+  print(node.tensor)  # This is how you access the underlying tensor.
+
+Edges
+-----
+Edges describe different computations of the underlying tensors in the tensor network. Each edge points to which axes of the tensors to do the computation. There are 3 basic kinds of edges in a tensor network:
+
+**Standard Edges**
+
+Standard edges are like any other edge you would find in an undirected graph. They connect 2 different nodes and define a dot product among the given vector spaces. In numpy terms, this edge defines a tensordot operation over the given axes.
+
+**Trace Edges**
+
+Trace edges connect a node to itself. To contract this type of edge, you take a trace of the matrix created by the two given axes.
+
+**Dangling Edge**
+
+Dangling edges are edges that only have one side point to a node, where as the other side is left “dangling”. These edges represent output axes or intermediate axes that have yet to be connected to other dangling edges. These edges are automatically created when adding a node to the network.
+
+.. code-block:: python
+
+  net = tensornetwork.TensorNetwork()
+  a = net.add_node(tf.eye(2))
+  b = net.add_node(tf.eye(2))
+  c = net.add_node(tf.eye(2))
+  # Dangling edges are automatically created at node creation. 
+  # We can access them this way.
+  dangling_edge = a.get_edge(1)
+  # This is the same as above
+  dangling_edge = a[1]
+  # Create a standard edge by connecting any two separate nodes together.
+  # We create a new edge by "connecting" two dangling edges.
+  standard_edge = net.connect(a[0], b[0]) 
+  # Create a trace edge by connecting a node to itself.
+  trace_edge = net.connect(c[0], c[1])
+
+Connecting Dangling Edges 
+-------------------------
+One common paradigm in building quantum circuits is to add computation based on edges rather than nodes. That is, it’s very common to want to connect two dangling edges rather than trying to keep track of node axes.
```

Comment:
```suggestion
One common paradigm in building quantum circuits, for example, is to add computation based on edges rather than nodes. That is, it’s very common to want to connect two dangling edges rather than trying to keep track of node axes.
```